scheduler/scheduler.go

```diff
@@ -140,6 +267,49 @@ type GroupedLIFOFilter interface {
 	HasConfig() bool
 }
 
+// Wait blocks until a request can be processed or needs to be rejected.
+// When it can be processed, calling done indicates that it has finished.
+// It is mandatory to call done() the request was processed. When the
+// request needs to be rejected, an error will be returned.
+func (fq *FifoQueue) Wait(ctx context.Context) (func(), error) {
+	f, err := fq.queue.Wait(ctx)
+	if fq.metrics != nil {
+		switch err {
+		case ErrQueueFull:
+			fq.metrics.IncCounter(fq.errorFullMetricsKey)
+		case ErrQueueTimeout:
+			fq.metrics.IncCounter(fq.errorTimeoutMetricsKey)
+		case ErrClientCanceled:
```

Comment:
Is `fallthrough` missing? Or maybe add a comment if not...

Reply:
no this is visible in 499, not really a scheduler filter issue. I add a comment